core/src/main/java/tc/oc/pgm/PGMConfig.java

```
@@ -146,6 +155,41 @@
     this.experiments = experiments == null ? ImmutableMap.of() : experiments.getValues(false);
   }
 
+  private static List<GitRepository> parseGitSources(
+      List<String> simpleRepos, List<Map<?, ?>> thoroughRepos) throws URISyntaxException {
+    List<GitRepository> gits = new ArrayList<>();
+
+    for (String repository : simpleRepos) {
+      gits.add(new GitRepository(repository));
+    }
+
+    for (Map<?, ?> repository : thoroughRepos) {
+      try {
+
+        URIish uri = new URIish((String) repository.get("uri"));
+
+        String branch = (String) repository.get("branch");
+
+        List<String> folders = (ArrayList<String>) repository.get("folders");
+
+        gits.add(new GitRepository(uri, branch, folders));
+      } catch (ClassCastException e) {
+        // Config parsing error, not catching this will break loading of the plugin if thrown
+        logger.log(
+            Level.WARNING,
+            "Config ERROR: A git repository was formatted incorrectly and will be ignored until restart");
+      } catch (URISyntaxException e) {
+        logger.log(
+            Level.WARNING,
+            "Config ERROR: A URI("
+                + e.getInput()
+                + ") was formatted incorrectly and will be ignored until restart",
+            e);
+      }
```

Instead of logging, throw the error, but wrapped as a `TextException`